src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2912,7 +2912,7 @@ private BoundCatchBlock BindCatchBlock(CatchClauseSyntax node, ArrayBuilder<Boun
 
                             diagnostics.Add(declaration.Type, useSiteDiagnostics);
                         }
-                        else if (TypeSymbol.Equals(previousType, Compilation.GetWellKnownType(WellKnownType.System_Exception), TypeCompareKind.ConsiderEverything2) &&
+                        else if (TypeSymbol.Equals(previousType, Compilation.GetWellKnownType(WellKnownType.System_Exception, recordUsage: false), TypeCompareKind.ConsiderEverything2) &&
```

Comment:
>ConsiderEverything2 [](start = 163, length = 19)

nit: when modifying a method, consider scrubbing ConsiderEverything2 (is ConsiderEverything the correct option?)

Reply:
> is ConsiderEverything the correct option?

The option used here has no relation with the purpose of this PR and I do not intend spending any time thinking about it under umbrella of this PR.

---
In reply to: [350908701](https://github.com/dotnet/roslyn/pull/40010#discussion_r350908701) [](ancestors = 350908701)

Reply:
I understand, but then I'm afraid we won't make progress on reviewing those incrementally, as every PR will likely raise similar pushback... Maybe worth discussing in team meeting (how do we make progress on those?), but unrelated to your PR. Thanks

---
In reply to: [350910134](https://github.com/dotnet/roslyn/pull/40010#discussion_r350910134) [](ancestors = 350910134,350908701)

Reply:
> but then I'm afraid we won't make progress on reviewing those incrementally, as every PR will likely raise similar pushback...

There is a PR that touches 90 files across wide range of features and there is a PR that focuses on specific feature where a comparison actually plays a role.  

---
In reply to: [350914019](https://github.com/dotnet/roslyn/pull/40010#discussion_r350914019) [](ancestors = 350914019,350910134,350908701)